arch/arm/core/aarch64/vector_table.S

```diff
@@ -62,11 +102,12 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 
 	/* Current EL with SP0 / Synchronous */
 	.align 7
+	z_arm64_enter_exc x0, x1
```

Comment:
I'm a bit torn about this change. I understand the rationale behind this change but I cannot say I particularly like it.

- First of all I understand that the most logical place to have `z_arm64_exit_exc` / `z_arm64_enter_exc` is in the vector table but they do look out of place (at least IMO)
- With this change you need to have one as a macro, the other as a function that is quite confusing if you don't know the details
- You are using `z_arm64_enter_exc` as preamble for each vector but there is no corresponding `z_arm64_exit_exc` that is still in the out-of-line code

So at the end of the day I wonder if this change is really better than having to waste space in each vector slot. All the remaining fixes are really good though.

Reply:
I think I'm forced to agree with you for the most part here, mainly because this isn't _actually_ saving space, because the vectors are already padded out.

Perhaps have both the entry and exit code in each of the vector slots?

The only space you could save here compared to having entry and exit in the vector slots is after the last vector slot in the table - but then you're replacing one exception exit with one exception exit, which isn't saving space. (Actually likely making the total space use _worse_, because the last vector is just a `b .` with no exception exit...)

It does mean that the actual exception handlers (`z_arm64_serror`, `_isr_wrapper`, etc) need to save/restore their LR, which isn't perfect.